funfact/model/_factorization.py

```diff
@@ -76,15 +80,52 @@ def forward(self):
         '''Evaluate the tensor expression the result.'''
         return self.tsrex | self._evaluator
 
-    def _get_elements(self, idx):
+    def _get_elements(self, key):
         '''Get elements at index of tensor expression.'''
-        new_idx = []
-        for i in idx:
+
+        # Generate full index list
+        full_idx = []
+        ellipsis_i = None
+        for i in key:
             if isinstance(i, int):
-                new_idx.append(slice(i, i+1))
+                if i != -1:
```

Comment:
Could `i` be a negative number other than -1?

Reply:
I think anything between `-size_of_dimension` and `size_of_dimension-1` could make sense?